### DISCUSSION TITLE
Permeability prediction.

### DISCUSSION THREAD

**Discussion by kizzle4shaw** [PMEAL/OpenPNM#2719] [Q&A]
I tried to extract pn of estallides carbonate using snow2 from pore spy, then I wanted to calculate absolute permeability, I followed the example provided since both of them are 3d domains. after running the stokes flow algorithm on the network I see this, no rate has been computed as seen on the image. More over i get warning and errors when I use the get_domain_area/length. What could something that I haven't considered perhaps. I believe the experts have worked with extracted networks before kindly help. 

 

![image](https://user-images.githubusercontent.com/32852010/229111332-42d88a58-c4e3-4af9-88df-fe030b36eb10.png)

**Comment by jgostick** [comment]
'pore.bc.rate' only gets assigned if you create a 'rate' boundary conditions.  You have only applied 'value' bcs.  To get that rate you have to ask the algorithm like `alg.rate(pores=Ps)`, where Ps is the pores where you applied the value BCs.

**Reply by jgostick** [reply]
Your question does make an interesting point though...maybe the algorithm should store the rate in each pore as a 'result', instead of having to compute it when asked for?  Note that the rate is most pores is 0 (if there is no reaction and it's steady-state).  The rate at the boundary pores is non-zero though, so users could do something like `alg['pore.rate'][Ps].sum()` to get the total rate passing through the pores `Ps`.

**Reply by kizzle4shaw** [reply]
If it calculates on assigned boundary, then it should automatically store it in bc.rate with the affected pores I think. 

My other question was  on the domain area calculation, I saw how you calculated on the Berea sample you added 6 I'm assuming it's because of the boundary width 3 during the snow extraction, but what about get_domain_area in the topotools, isn't it supposed to do it automatically?

**Reply by ma-sadeghi** [reply]
I don't think we should populate `bc.rate` automatically. `bc.rate` conveys that a rate boundary condition has been assigned. But maybe storing the rates on the algorithm object is not a bad idea.

**Reply by jgostick** [reply]
No, not ``'pore.bc.rate'``, just `'pore.rate'`.  Just as we add 'pore.concentration', for e.g.

**Reply by ma-sadeghi** [reply]
There's one question to be answered: how often should it be updated? Let's say we change the conductance values, upon regenerating models should we update pore.rate as well? Otherwise users might get into the habit of using pore.rate instead of actually calling rate(), which might not always up to date.

**Comment by jgostick** [comment]
Let's move this discussion to the issue/feature request #2720
